contributors/design-proposals/federated-placement-policy.md

```diff
@@ -0,0 +1,264 @@
+    "kind": "ReplicaSet",
+    "metadata": {
+      "annotations": {
+        "eu-jurisdiction-required": "true",
+        "pci-compliance-level": "2"
+      },
+      "creationTimestamp": "2017-01-23T16:25:14Z",
+      "generation": 1,
+      "labels": {
+        "app": "nginx-eu"
+      },
+      "name": "nginx-eu",
+      "namespace": "default",
+      "resourceVersion": "364993",
+      "selfLink": "/apis/extensions/v1beta1/namespaces/default/replicasets/nginx-eu",
+      "uid": "84fab96d-e188-11e6-ac83-0a580a54020e"
+    },
+    "spec": {
+      "replicas": 4,
+      "selector": {...},
+      "template": {...},
+    }
+  }
+}
+```
+
+#### Example Policy Engine Response:
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+```
+
+```json
+{
+  "result": {
+    "federation.kubernetes.io/replica-set-preferences": {
+      "clusters": {
+        "gce-europe-west1": {
+          "weight": 1
+        },
+        "gce-europe-west2": {
+          "weight": 1
+        }
+      },
+      "rebalance": true
+    }
+  }
+}
+```
+
+#### Error Handling
+
+The admission controller is designed to **fail closed**. Request handling may
+fail because of:
+
+Serialization errors Request timeouts or other network errors Unexpected errors
+from the policy engine
+
+If the administrator has not defined a policy yet, the response from the policy
+engine indicates no annotations are defined. The admission controller does not
+treat this as an error, i.e., it admits the request.
+
+#### Configuration
+
+The admission controller requires configuration for the OPA endpoint:
+
+```
+{
+  “opa”: {
+    “baseURL”: “http://opa.federation.svc.cluster.local:8181/v1”,
+    “annotationsPath”: “/data/io/k8s/federation/annotations”,
+    “ignoreUserNames”: [“opa”]
+  }
+}
+```
+
+The configuration also contains a list of usernames that identify senders whose
+requests should be ignored. This is used to avoid introducing a loop when a
+remediation component is deployed.
+
+The configuration file is provided to the federation-apiserver with the
+`--admission-control-config-file` command line argument.
+
+The admission controller is enabled in the federation-apiserver by providing the
+`--admission-control` command line argument. E.g.,
+`--admission-control=AlwaysAdmit,OPA`.
+
+### 2. Remediation of Resource Placement
+
+When policy changes or the environment in which resources are deployed changes
+(e.g. a cluster’s PCI compliance rating gets up/down-graded), resources might
+need to be moved for them to obey the placement policy. Sometimes operators may
+decide to remediate manually, but sometimes they may want Kubernetes to
+remediate automatically.
+
+To automatically reschedule resources onto desired clusters, we introduce the
+sidecar (**opa-kube-sync**) to receive notifications from OPA that indicate
+changes in placement that would bring resources back into policy compliance. The
+sidecar reacts to notifications from OPA by updating the resources in the
```

Comment:
OK. I will add more detail to this section to explain the mechanics.

> Note also that there is the potential for confusion (by readers of your doc) between this form of resource movement, and the Rescheduling Algorithm described in the above doc, so it might be worth adding a brief clarification in that regard here.

I'll update the section to clarify that the policy engine only sends the remediator the **desired value** for the annotations and that the actual rebalancing is still handled by the algorithm in the controller. Thanks!